cloudinit/config/schema.py

```diff
@@ -785,25 +794,39 @@ def validate_cloudconfig_schema(
             )
             if prop_match:
                 path = prop_match["name"]
-        problem = (SchemaProblem(path, schema_error.message),)
         if isinstance(
             schema_error, SchemaDeprecationError
         ):  # pylint: disable=W1116
-            deprecations += problem
+            if (
+                "devel" != features.DEPRECATION_INFO_BOUNDARY
+                and Version.from_str(schema_error.version) >
+                Version.from_str(features.DEPRECATION_INFO_BOUNDARY)
+            ):
+                info_deprecations.append(
+                    SchemaProblem(path, schema_error.message)
```

Comment:
The original code created a tuple with a single element and then appended this single element to a list:

```python
problem = (SchemaProblem(path, schema_error.message),)
...
deprecations += problem
```

which takes some extra mental gymnastics to realize the end goal is to append to a list

```python
deprecations.append(SchemaProblem(path, schema_error.message))
```

I consider the legibility difference between:

```python
if <blah>:
    info_deprecations.append(SchemaProblem(path, schema_error.message))
else:
    deprecations.append(SchemaProblem(path, schema_error.message))
```

and

```python
problem = SchemaProblem(path, schema_error.message)
if <blah>:
    info_deprecations.append(problem)
else:
    deprecations.append(problem)
```
to be negligible. The second has less total calls to grok, but the first has an extra variable to track, and since the calls are simple and identical it's quick realize that they do the same thing.

> what's the perspective on dropping the local var declaration 'problem =' 

It's really just a style preference. When I find differences like this negligible I typically lean towards the option that involves assigning fewer variables for a few of reasons:

1) less state / context to track -> easier to mentally follow
2) easier to refactor into simpler / cleaner code in the future
3) less variable names that I need to think up (obviously doesn't apply in this case because pre-existing code)
4) less lines of code to read / maintain

While these reasons reasons seem bit trivial, the impacts add up as code size and complexity grows.

I suppose I could have done this PR without making this change, but the actual reason that this happened due to an earlier iteration subclassing a different type and having to move the assignment within the `if isinstance()` call which necessitated modifying this variable anyways.

Reply:
Good intent/reasoning here thanks.